test/medley/core_test.cljc

```diff
@@ -355,6 +355,26 @@
                       [1 3 5 6 8 9])
            [[1 3 5] [6]]))))
 
+(deftest test-partition-with
+  (testing "sequences"
+    (is (= (m/partition-with = nil) '()))
```

Comment:
We likely also want to test the case where there is exactly one element in the collection, as this is a special case (i.e. `(x)` is always `((x))` regardless of the predicate).